samples/modules/chre/src/main.cpp

```diff
@@ -28,6 +29,12 @@ int main(void)
 	printk("Hello CHRE!\n");
 
 	k_msleep(500);
+	/*
+	 * Flush all log messages that resulted from initialization to avoid
+	 * getting them mingled with those printk messages below.
+	 */
+	while (LOG_PROCESS()) {
```

Comment:
it would be better to sleep and let logging thread do the work. Logging is not ready to handle processing from 2 contexts. It that interrupts logging thread something may get corrupted.

See how ztest is backing off until all logging messages are processed: https://github.com/zephyrproject-rtos/zephyr/blob/cd3c3cf6ebc136e8672220382a0c59622b51531b/subsys/testsuite/ztest/src/ztest.c#L399